util/src/main/java/com/redhat/lightblue/util/MemoryMonitor.java

```diff
@@ -0,0 +1,85 @@
+package com.redhat.lightblue.util;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Utility class for tracking memory usage for given type (assuming it's an immutable value).
+ * You can set thresholds and have it execute callbacks when they are exceeded. See {@link MemoryMonitorTest} for usage.
+ *
+ * @author mpatercz
+ *
+ * @param <T>
+ */
+public class MemoryMonitor<T> {
+
+    @FunctionalInterface
+    public static interface SizeCalculator<T> {
+        public int size(T obj);
+    }
+
+    @FunctionalInterface
+    public static interface ThresholdExceededCallback<T> {
+        public void fire(int currentSizeB, int thresholdB, T obj) throws RuntimeException;
+    }
+
+    public static class ThresholdMonitor<T> {
+
+        final int thresholdB;
+
+        boolean fired = false;
+
+        final ThresholdExceededCallback<T> event;
+
+        public ThresholdMonitor(int thresholdB, ThresholdExceededCallback<T> event) {
+            super();
+            this.thresholdB = thresholdB;
+            this.event = event;
+        }
+
+    }
```

Comment:
Not sure I understand. You're proposing to add 2 more methods with the logic currently implemented in checkThresholdMonitors?